src/server.c

```diff
@@ -1009,19 +1010,49 @@ static void on_connection(void* obj)
 	free(client);
 }
 
-static int bind_address(const char* name, int port)
+static int getaddrinfo_unix(const char *name, const char *service, struct addrinfo *hints, struct addrinfo **res)
+{
+	struct sockaddr_un *addr;
+
+	if (hints->ai_family != AF_UNIX)
+		return getaddrinfo(name, service, hints, res);
+
+	if (strlen(name) >= sizeof(addr->sun_path))
+		return -1;
+	if (!(addr = malloc(sizeof(*addr))))
+		return -1;
+	if (!(*res = calloc(1, sizeof(**res)))) {
+		free(addr);
+		return -1;
+	}
+
+	addr->sun_family = AF_UNIX;
+	strcpy(addr->sun_path, name);
+
+	(*res)->ai_family = AF_UNIX;
+	(*res)->ai_socktype = SOCK_STREAM;
+	(*res)->ai_addrlen = sizeof(*addr);
+	(*res)->ai_addr = (struct sockaddr *)addr;
+
+	return 0;
+}
+
+static int bind_address(const char* name, int port, bool unix_)
```

Comment:
shoehorning unix sockets into this function is a bit messy. There should be two functions instead: `bind_address_tcp` and `bind_address_unix`.